ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,935 @@
+            return timestamps.dt.tz_convert(tz)
+        else:
+            return timestamps.dt.tz_localize(tz)
+
+    raise TypeError("Don't know how to cast {} to {}".format(from_type, type))
+
+
+@execute_node.register(ops.Cast, dd.Series, dt.Date)
+def execute_cast_series_date(op, data, type, **kwargs):
+    arg = op.args[0]
+    from_type = arg.type()
+
+    if from_type.equals(type):
+        return data
+
+    # TODO - we return slightly different things depending on the branch
+    # double check what the logic should be
+
+    if isinstance(from_type, dt.Timestamp):
+        return data.dt.normalize()
+
+    if from_type.equals(dt.string):
+        # TODO - this is broken
+        datetimes = data.map_partitions(
+            to_datetime,
+            infer_datetime_format=True,
+            meta=(data.name, 'datetime64[ns]'),
+        )
+        try:
+            datetimes = datetimes.dt.tz_convert(None)
+        except TypeError:
+            pass
+        # TODO - we are getting rid of the index here
+        return datetimes.dt.normalize()
+
+    if isinstance(from_type, dt.Integer):
+        return data.map_partitions(
+            to_datetime, unit='D', meta=(data.name, 'datetime64[ns]')
+        )
+
+    raise TypeError("Don't know how to cast {} to {}".format(from_type, type))
+
+
+@execute_node.register(ops.SortKey, dd.Series, bool)
+def execute_sort_key_series_bool(op, data, ascending, **kwargs):
+    return data
+
+
+@execute_node.register(ops.Negate, dd.Series)
+def execute_series_negate(op, data, **kwargs):
+    return data.mul(-1)
+
+
+@execute_node.register(ops.Negate, SeriesGroupBy)
+def execute_series_group_by_negate(op, data, **kwargs):
+    return execute_series_negate(op, data.obj, **kwargs).groupby(
+        data.grouper.groupings
+    )
+
+
+def call_numpy_ufunc(func, op, data, **kwargs):
+    if data.dtype == np.dtype(np.object_):
+        return data.apply(
+            functools.partial(execute_node, op, **kwargs),
+            meta=(data.name, "object"),
+        )
+    return func(data)
+
+
+@execute_node.register(ops.UnaryOp, dd.Series)
+def execute_series_unary_op(op, data, **kwargs):
+    function = getattr(np, type(op).__name__.lower())
+    return call_numpy_ufunc(function, op, data, **kwargs)
+
+
+@execute_node.register((ops.Ceil, ops.Floor), dd.Series)
+def execute_series_ceil(op, data, **kwargs):
+    return_type = np.object_ if data.dtype == np.object_ else np.int64
+    func = getattr(np, type(op).__name__.lower())
+    return call_numpy_ufunc(func, op, data, **kwargs).astype(return_type)
+
+
+def vectorize_object(op, arg, *args, **kwargs):
+    # TODO - this works for now, but I think we can do something much better
+    func = np.vectorize(functools.partial(execute_node, op, **kwargs))
+    out = dd.from_array(func(arg, *args), columns=arg.name)
+    out.index = arg.index
+    return out
+
+
+@execute_node.register(
+    ops.Log, dd.Series, (dd.Series, numbers.Real, decimal.Decimal, type(None))
```

Comment:
can you group numeric funcs (and put a coment for a sub-section), similar for other things can group together; makes it much easier to find / read this file.